Remove unused includes from regression tests

tautschnig opened the pull request:
This cleanup avoids giving the false impression we would be testing any
functions declared in those header files.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
Remove unused includes from regression tests

This cleanup avoids giving the false impression we would be testing any
functions declared in those header files, which in turn simplifies moving tests
that exercise the internal modelling library to a separate folder.

martin-cs commented:
Am curious to know how you found these (if automated; is this all of them?) but possibly even more curious to know why this causes failures in the unit tests...

tautschnig committed:
clang-format whitespace cleanup in regression test

clang-format was complaining about lines I hadn't touched, cleaning them up
anyhow.

tautschnig commented:
@martin-cs This is now fixed: one of the includes provided `NULL`...

allredj reviewed (commented):
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 10409de86debf1812d4af5270459d7aaeee8e11d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89574972
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.
